olive/evaluator/olive_evaluator.py

```diff
@@ -236,12 +234,28 @@ def generate_metric_user_config_with_model_io(metric: Metric, model: "OliveModel
                 "Model input shapes are not static. Cannot use inferred input shapes for creating dummy data. This will"
                 " cause an error when creating dummy data for tuning."
             )
-        if io_config and not metric.user_config.input_names and not metric.user_config.input_shapes:
+
+        if metric.data_config:
```

Comment:
I don't think this is a good idea. If a user chose the type as `dummydatacontainer` and overrode the load_dataset parameter (like how the hf container is overridden in the resnet examples) these params are not relevant. It also doesn't sound like a good idea for the evaluator to update a pre-existing container. 

Since the only use case right now is the squeezenet example. Why not just remove the `data_config` value passed in the perf tuning config. the perf tuning pass is able to create it's own data config just like this if the data config is not provided.

Reply:
Updated.